From cbd3533428f34b73bd59c31b264bfedda495514b Mon Sep 17 00:00:00 2001
From: Meron Cohen <mecohen@habana.ai>
Date: Tue, 13 May 2025 22:18:06 +0300
Subject: [PATCH] [SW-227605] g2e-pcie: add is_bis_e_compliant api

For G2E PCIe card, there is to support HBM RLs.
Temporarily, the solution is not part of the hl_rl driver.
HBM RLs are configured during initializtion of arcmgmt.
It is a common code, so there is to verify it is done
for G2E PCIe card only. That's why this API is added.

Change-Id: I150a07cf26d6b5d050fe7a342b5c7bd44c754314
Signed-off-by: Meron Cohen <mecohen@habana.ai>
---

diff --git a/drivers/misc/common_utils/gaudi/gaudi2/common_utils.c b/drivers/misc/common_utils/gaudi/gaudi2/common_utils.c
index e91908c..2226ae3 100644
--- a/drivers/misc/common_utils/gaudi/gaudi2/common_utils.c
+++ b/drivers/misc/common_utils/gaudi/gaudi2/common_utils.c
@@ -181,6 +181,14 @@
 	return false;
 }
 
+bool is_bis_e_compliant_device(void)
+{
+	if (!bis_compliant_cache.is_valid)
+		init_bis_cache();
+
+	return bis_compliant_cache.bis_rev == GAUDI2_BIS_REV_E;
+}
+
 bool hl_dev_is_valid_power(uint64_t value)
 {
 	if (value < GAUDI2_MAX_POWER_MIN_VAL_MW) {
diff --git a/drivers/misc/common_utils/gaudi/gaudi_utils.c b/drivers/misc/common_utils/gaudi/gaudi_utils.c
index 666316b..dd3a9e2 100644
--- a/drivers/misc/common_utils/gaudi/gaudi_utils.c
+++ b/drivers/misc/common_utils/gaudi/gaudi_utils.c
@@ -110,6 +110,11 @@
 	return false;
 }
 
+__weak bool is_bis_e_compliant_device(void)
+{
+	return false;
+}
+
 __weak bool hl_dev_is_valid_power(uint64_t value)
 {
 	return true;
diff --git a/include/zephyr/drivers/misc/common/common_utils.h b/include/zephyr/drivers/misc/common/common_utils.h
index 8ce0506..ae38715 100644
--- a/include/zephyr/drivers/misc/common/common_utils.h
+++ b/include/zephyr/drivers/misc/common/common_utils.h
@@ -276,6 +276,7 @@
 bool hl_is_xdpp1100_present(void);
 bool hl_is_adc121c027_present(void);
 bool is_bis_compliant_device(void);
+bool is_bis_e_compliant_device(void);
 bool hl_is_ina236_present(uint16_t addr);
 bool hl_dev_is_valid_power(uint64_t value);
 bool hl_is_security_enabled(void);
